lib/src/transformer/declaration_parsing.dart

```diff
@@ -53,11 +53,21 @@ class ParsedDeclarations {
       key_stateMixin:        <CompilationUnitMember>[],
     };
 
+    var privatePropsAndStateClassPattern = new RegExp(r'^(?:\@Props\(\)\s*|\@State\(\)\s*)(?:abstract\s)?class\s(_\$\w*?)\s');
+
     unit.declarations.forEach((CompilationUnitMember member) {
       member.metadata.forEach((annotation) {
         var name = annotation.name.toString();
+        var match = privatePropsAndStateClassPattern.firstMatch(member.toString());
```

Comment:
Talked about this offline, but we'll need to do a nested for-loop check here to find the public-pair instead of using a regex so that we don't have to account for all the possible variations (annotation params, extra annotations, comments, etc.)